src/praisonai/tests/unit/integrations/test_context_files.py

```
@@ -98,3 +102,104 @@ def test_walk_up_disabled_skips_global_file(project, monkeypatch, tmp_path):
+def test_sibling_without_instructions_attaches_nothing_new(monorepo):
+    root, _foo, bar = monorepo
+    (root / "AGENTS.md").write_text("ROOT RULES")
+
+    attacher = PathContextAttacher(already_loaded="ROOT RULES")
+    out = attacher.attach_for_path(bar / "main.py")
+    # bar has no own AGENTS.md and root was already loaded -> nothing new.
+    assert out == ""
+
+
+def test_dedup_against_already_loaded_up_front(monorepo):
+    root, foo, _bar = monorepo
```

Ruff (0.15.20)

[warning] Unpacked variable `root` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)